File: Taki/TakiServer/Card.cpp
#include "Card.h"


Card::Card()
{
}

Card::Card(char color, char type)
{
	_type = type;
	_color = color;
}

Card::~Card()
{
}

string Card::to_string()
{
	return { _type, _color };
}

char Card::getColor() const
{
	return _color;
}

char Card::getType() const
{
	return _type;
}

void Card::setType(char type)
{
	_type = type;
}

void Card::setColor(char color)
{
	_color = color;
}

bool Card::operator==(const Card &other) const
{
	return _type == other._type && _color == other._color;
}

File: Taki/TakiClient/Card.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace newGUI_Taki
{
    class Card
    {
        private char _type;
        private char _color;
        string to_string()
        {
            char[] a = {_type,_color};
	        return a.ToString();
        }

        char getColor()
        {
            return _color;
        }

        char getType()
        {
            return _type;
        }

        void setType(char type)
        {
            _type = type;
        }

        void setColor(char color)
        {
            _color = color;
        }

    }
}
